src/app/page.tsx

```diff
@@ -1,95 +1,87 @@
-        </a>
+export default function Home() {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [articles, setArticles] = useState<ArticleAPI[]>([]);
+  const [searchQuery, setSearchQuery] = useState<string | null>(null);
+  const [isSearchPerformed, setIsSearchPerformed] =
+    useState<boolean>(false);
 
-        <a
-          href="https://nextjs.org/learn?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
-          className={styles.card}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <h2>
-            Learn <span>-&gt;</span>
-          </h2>
-          <p>Learn about Next.js in an interactive course with&nbsp;quizzes!</p>
-        </a>
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      try {
+        const response = await fetchNews();
+        setArticles(response.articles);
+        setIsLoading(false);
+      } catch (err) {
+        console.error(err);
+        setIsLoading(false);
+      }
+    }
+    fetchData();
+  }, []);
 
-        <a
-          href="https://vercel.com/templates?framework=next.js&utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
-          className={styles.card}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <h2>
-            Templates <span>-&gt;</span>
-          </h2>
-          <p>Explore the Next.js 13 playground.</p>
-        </a>
+  useEffect(() => {
+    async function performSearch() {
+      if (searchQuery) {
+        setIsLoading(true);
+        try {
+          const response = await fetchSearch({
+            request: searchQuery,
+          });
+          setArticles(response.articles);
+          setIsLoading(false);
+          setIsSearchPerformed(true);
+        } catch (err) {
+          setIsLoading(false);
+          console.error(err);
+        }
+      }
+    }
+    performSearch();
+  }, [searchQuery]);
 
-        <a
-          href="https://vercel.com/new?utm_source=create-next-app&utm_medium=appdir-template&utm_campaign=create-next-app"
-          className={styles.card}
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          <h2>
-            Deploy <span>-&gt;</span>
-          </h2>
-          <p>
-            Instantly deploy your Next.js site to a shareable URL with Vercel.
-          </p>
-        </a>
+  return (
+    <main className={Styles.main}>
+      <div className={Styles.center}>
+        <SearchPanel
+          setSearchQuery={setSearchQuery}
+          placeholder='Search news ...'></SearchPanel>
+        {searchQuery && <h2>Results of : '{searchQuery}'</h2>}
+        {isLoading ? (
+          <Image
+            width={300}
+            height={300}
+            src='/loading.gif'
+            alt='Loading'
+            className={Styles.spinner}
+          />
+        ) : null}
+        {isSearchPerformed && articles.length < 1 && (
```

Comment:
Instead of `isSearchPerformed && articles.length < 1 `would it better to have en error state?

Reply:
But if articles.length < 1 ,there is no error